components/concept/engine/src/main/java/mozilla/components/concept/engine/prompt/Choice.kt

```diff
@@ -0,0 +1,63 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.concept.engine.prompt
+
+import android.os.Parcel
+import android.os.Parcelable
+
+/**
+ * Value type that represents a select option, optgroup or menuitem html element.
+ *
+ * @property id of the option, optgroup or menuitem.
+ * @property enable indicate if item should be selectable or not.
+ * @property label The label for displaying the option, optgroup or menuitem.
+ * @property selected Indicate if the item should be pre-selected.
+ * @property isASeparator Indicating if the item should be a menu separator (only valid for menus).
+ * @property children Sub-items in a group, or null if not a group.
+ */
+data class Choice(
+    val id: String,
+    var enable: Boolean = true,
+    var label: String,
+    var selected: Boolean = false,
```

Comment:
Do those need to be mutable?

Reply:
Yes, I'm mutating them when is multiple choice dialog, instead of creating new new objects.

https://github.com/mozilla-mobile/android-components/blob/ffdd513f100c0c119940694f728d5279d3cc4144/components/feature/session/src/main/java/mozilla/components/feature/session/prompt/ChoiceAdapter.kt#L125

![image](https://user-images.githubusercontent.com/773158/49029450-a87b0380-f172-11e8-8ab1-e720e351d1cc.png)